src/Shiny.BluetoothLE/Platforms/Uwp/GattService.cs

```diff
@@ -47,23 +49,24 @@ public override IObservable<IGattCharacteristic> GetKnownCharacteristics(params
             });
 
 
-        IObservable<IGattCharacteristic> characteristicOb;
         public override IObservable<IGattCharacteristic> DiscoverCharacteristics()
         {
-            this.characteristicOb = this.characteristicOb ?? Observable.Create<IGattCharacteristic>(async ob =>
+            var discoverCharacteristics = new Subject<IGattCharacteristic>();
+            Task.Run(async () => 
```

Comment:
Why task.run with the await below?

Reply:
Because Windows.Devices.Bluetooth.GenericAttributeProfile.GattDeviceService.GetCharacteristicsAsync(BluetoothCacheMode cacheMode) is awaitable, this looks the simpliest pattern.

Reply:
It can be wrapped in an observable a bit cleaner.  This is fine though.  I'll update once the PR is received.